docs/adr/0024-upgrade-templating-toolset-of-auth-server-emails.md

```diff
@@ -0,0 +1,177 @@
+- Plans around integration involving feature flagging and the QA process
+
+## Decision Outcomes
+
+### Templating and Styling
+
+Chosen option: "Option C - Use EJS and MJML, using CSS options offered by MJML", because:
+
+- HTML email has a lot of quirks - MJML shifts the burden of maintaining solutions for these off of FxA engineers now and in the future
+- While we use React and Tailwind in other parts of FxA, React is heavy for an email solution since no state is involved, component reuse across FxA would likely be very minimal, and it involves a more complex build setup than EJS
+- MJML helps significantly with responsiveness in emails, reducing time spent developing templates and making future email redesigns easier
+
+### Email Previewing
+
+Chosen option: "Option B - Use Storybook", because:
+
+- It is easy to setup and provides us the flexibility to preview all of the different email states together.
+- We have a Storybook deployment already in place and hooking it up for `fxa-auth-server` will ensure consistency.
+- Easy to test and perform QA without needing to touch the codebase.
+- Support for CSS (whether it's plain old CSS, TailwindCSS or something else)
+
+Note: Along with using storybook to view the various states of the templates, we are also planning on continued support of [Maildev](https://www.npmjs.com/package/maildev) since it has applications beyond just previewing emails.
+
+### Other
+
+**Plaintext files**: We'll use [html-to-text](https://github.com/html-to-text/node-html-to-text) to automatically generate plaintext versions of templates rendered to HTML. There may be a scenario in which the automatically-generated plaintext version does not look exactly how we'd like, in which case we can look into exporting both an MJML _and_ plaintext version of the email from the template file.
+
+**Localization**: We will upgrade the localization tool from GetText to [Fluent](https://github.com/projectfluent/fluent.js) since it's preferred by the l10n team and other FxA front-ends are using it. With our chosen templating option we can make use of Fluent's [`@fluent/dom`](https://github.com/projectfluent/fluent.js/tree/master/fluent-dom) package.
+
+[**Integration & QA**: During development templates will be marked as being part of a release group. This could be in the form of a mapped list of template names or some variable associated with the template file. Each group will have a corresponding environment variable flag. When this release group’s flag is enabled any templates that fall under it will be used when generating an email template in that environment; until a template's corresponding release group is enabled it is not used and the old/current template will continue to be served. Release groups will be initially enabled in staging until QA has had an opportunity to thoroughly test and evaluate each template in the group, after which they can be incrementally enabled in production.] **Vijay to edit**
```

Comment:
> Vijay to edit

To give us the most flexibiity we can add a new [auth-server configuration](https://github.com/mozilla/fxa/blob/4c90c1ca6afed04d550690de4b7248c7906adeb4/packages/fxa-auth-server/config/index.ts#L1486) value that can control which users email are supported and which email templates are supported for mjml. The auth-server can then expose a [feature flag method](https://github.com/mozilla/fxa/blob/cc595fc2b44bc1e15857c016539622a6a750bd5d/packages/fxa-auth-server/lib/features.js#L23) to check this value and then render the correct [template](https://github.com/mozilla/fxa/blob/f29f73e33b8b6ed03e4afa5442fcb0ee9abfe17d/packages/fxa-auth-server/lib/senders/email.js#L433-L432).